ui/app/AppLayouts/Shell/ShellGrid.qml

```diff
@@ -53,160 +54,172 @@ StatusGridView {
         pinned             [bool]   - whether the item is pinned in the UI
         timestamp          [int]    - timestamp of the last user interaction with the item
     **/
+    property alias model: repeater.model
 
-    property int cellSize: 160
-    property int cellPadding: Theme.padding
+    property int delegateWidth: 160
+    Behavior on delegateWidth {
+        PropertyAnimation { duration: Theme.AnimationDuration.Fast }
+    }
+    property int delegateHeight: 160
+    Behavior on delegateHeight {
+        PropertyAnimation { duration: Theme.AnimationDuration.Fast }
+    }
 
     signal itemActivated(string key, int sectionType, string itemId)
     signal itemPinRequested(string key, bool pin)
     signal dappDisconnectRequested(string dappUrl)
 
-    cellWidth: cellSize + cellPadding
-    cellHeight: cellWidth
-
-    ScrollBar.vertical: StatusScrollBar {
-        policy: ScrollBar.AsNeeded
-        visible: resolveVisibility(policy, root.height, root.contentHeight)
-    }
-
-    delegate: Loader {
-        required property int index
-        required property var model
-
-        sourceComponent: {
-            switch (model.sectionType) {
-            case Constants.appSection.profile:
-                return settingsDelegate
-            case Constants.appSection.community:
-                return communityDelegate
-            case Constants.appSection.wallet:
-                return walletDelegate
-            case Constants.appSection.chat:
-                return chatDelegate
-            case Constants.appSection.dApp:
-                return dappDelegate
-            default:
-                console.warn("Unhandled ShellGridItem delegate for sectionType:", model.sectionType)
+    padding: 0
+    rightPadding: Theme.padding
+
+    ColumnLayout {
+        width: root.width - root.rightPadding
+
+        Flow {
+            // calculate a tight bounding box, and then horizontally center over the scrollview width
+            readonly property int delegateCountPerRow: Math.trunc(parent.width / (root.delegateWidth + spacing))
+            Layout.preferredWidth: (delegateCountPerRow * root.delegateWidth) + (spacing * (delegateCountPerRow - 1))
+            Layout.alignment: Qt.AlignHCenter
+
+            spacing: Theme.padding
+
+            Repeater {
```

Comment:
Everything below has no other change; just the indentation changes due to it being wrapped in ScrollView/ColumnLayout/Flow